### PR TITLE
Add unit tests for indices.recovery.max_bytes_per_sec default values

### PR DESCRIPTION
**Pull request by tlrx**
`indices.recovery.max_bytes_per_sec` has a default value that depends on multiple criteria that are well documented but not unit tested. This pull request introduces unit tests that verifies the current behavior so that future changes like #82819 are less likely to break things.

### CODE REVIEW

**Comment by tlrx** [comment]
@elasticmachine run elasticsearch-ci/bwc

**Review by idegtiarenko — APPROVED** [review]

File: server/src/test/java/org/elasticsearch/indices/recovery/RecoverySettingsTests.java [hunk 1]
```diff
@@ -89,4 +103,199 @@ public void testMaxConcurrentSnapshotFileDownloadsPerNodeIsValidated() {
...
+                .getMaxBytesPerSec(),
+            equalTo(DEFAULT_MAX_BYTES_PER_SEC)
+        );
+    }
+
+    public void testMaxBytesPerSecOnDataNodeWithIndicesRecoveryMaxBytesPerSec() {
+        final ByteSizeValue random = randomByteSizeValue();
+        assertThat(
+            "Data nodes that are not dedicated to cold/frozen should use the defined rate limit when set",
+            nodeRecoverySettings().withRole(randomFrom("data", "data_hot", "data_warm", "data_content"))
```
Comment (henningandersen) [inline server/src/test/java/org/elasticsearch/indices/recovery/RecoverySettingsTests.java:143]: Can we randomly add either `data_cold` or `data_frozen` (or both) to the set of roles here?

**Review by henningandersen — APPROVED** [review]
LGTM.

File: server/src/main/java/org/elasticsearch/indices/recovery/RecoverySettings.java [hunk 2]
```diff
@@ -46,36 +45,61 @@
...
+                : JavaVersion.current();
+            if (javaVersion.compareTo(JavaVersion.parse("14")) < 0) {
                 // prior to JDK 14, the JDK did not take into consideration container memory limits when reporting total system memory
-                return defaultMaxBytesPerSec.getStringRep();
+                return DEFAULT_MAX_BYTES_PER_SEC.getStringRep();
             }
-            final ByteSizeValue totalPhysicalMemory = new ByteSizeValue(OsProbe.getInstance().getTotalPhysicalMemorySize());
+            final ByteSizeValue totalPhysicalMemory = TOTAL_PHYSICAL_MEMORY_OVERRIDING_TEST_SETTING.exists(s)
+                ? TOTAL_PHYSICAL_MEMORY_OVERRIDING_TEST_SETTING.get(s)
+                : new ByteSizeValue(OsProbe.getInstance().getTotalPhysicalMemorySize());
```
[the remark this answers is not in the document]
Comment (tlrx) [reply server/src/main/java/org/elasticsearch/indices/recovery/RecoverySettings.java]: Sure

File: server/src/test/java/org/elasticsearch/indices/recovery/RecoverySettingsTests.java (hunk 1 above)
Comment (tlrx) [reply server/src/test/java/org/elasticsearch/indices/recovery/RecoverySettingsTests.java]: You spotted it :)

**Comment by tlrx** [comment]
Thanks Ievgen & Henning

**Pull request merged**